internal/audit/file.go

```
@@ -77,16 +106,35 @@ func newFile(dir string, engine *crypto.Engine, logger log.Logger) (*file, error
-// open generates a fresh file name in dir and opens it for append.
-func open(dir string) (*os.File, string, error) {
+// open generates a fresh file name in dir, opens it for append, and writes the
+// self-describing header so every rotated segment carries its own key metadata.
+func open(dir string, keyMode byte, fingerprint [16]byte) (*os.File, string, error) {
 	path := filepath.Join(dir, fileName(dir))
 	osFile, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0o600)
 	if err != nil {
 		return nil, "", err
 	}
+	if err := writeHeader(osFile, keyMode, fingerprint); err != nil {
+		osFile.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `osFile.Close` is not checked

(errcheck)